uvicorn/protocols/websockets/websockets_sansio_impl.py

```diff
@@ -0,0 +1,389 @@
+import asyncio
+import logging
+import sys
+import typing
+from asyncio.transports import BaseTransport, Transport
+from http import HTTPStatus
+from urllib.parse import unquote
+
+import websockets
+from websockets.extensions.permessage_deflate import ServerPerMessageDeflateFactory
+from websockets.frames import Frame
+from websockets.http11 import Request, Response
+from websockets.server import ServerProtocol
+
+from uvicorn.config import Config
+from uvicorn.logging import TRACE_LOG_LEVEL
+from uvicorn.protocols.utils import (
+    get_local_addr,
+    get_path_with_query_string,
+    get_remote_addr,
+    is_ssl,
+)
+from uvicorn.server import ServerState
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
```

Comment:
I guess, this condition can be removed as uvicorn requirement for python is atleast 3.8. right?

Reply:
yes